bsg_misc/bsg_round_robin_arb.py

```diff
@@ -213,6 +214,8 @@ def print_reset_on_logic(reqs_w):
       end else if( reset_on_sr_p ) begin: reset_on_last_n_gen
         last_n = reset_on_sr? (inputs_p-2) :
                ( yumi_i     ?tag_o : last_r );  
+      end else if( hold_on_valid_p ) begin: hold_on_last_n_gen
+        last_n = yumi_i ? tag_o : v_o ? (tag_o-1'b1) : last_r;
```

Comment:
I guess `tag_o-1'b1` may not be valid when inputs_p is non-pow2? For example, when `inputs_p==5` and `tag_o==3'b000`, then `tag_o-1'b1==3'b111`, and the state machine enters default state ,making `sel_one_hot_n= {5{1'bx}}`... Maybe we should do the following?
```
last_n = yumi_i ? tag_o : v_o ? ((tag_o == '0)? (lg_inputs_p)'(inputs_p-1) : (tag_o-1'b1)) : last_r;
```
This error is not triggered in the testbench, because there is not data injected into router from proc, so tag_o is never set to 0...

Reply:
Good catch!

Reply:
Fixed!